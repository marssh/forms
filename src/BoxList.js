import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import { NewBoxForm } from './NewBoxForm';
import { Box } from './Box'


export function BoxList() {
  const [boxes, setBoxes] = useState([]);

  const addItem = box => {
    setBoxes((currBoxes) => {
      box.id = uuid();
      box.width = +box.width;
      box.height = +box.height;
      const totalBox = [...currBoxes, box];
      return totalBox;
    })
  }

  const deleteItem = id => {
    setBoxes(currBoxes => currBoxes.filter(el => el.id !== id));
  }


  return (
    <div>
      <NewBoxForm addItem={addItem} />
      {boxes
        .map(({ backgroundColor, height, width, id}) =>
          <Box key={id} id={id} style={{ backgroundColor, height, width }} deleteItem={deleteItem}/>
        )}
    </div>
  )
}


