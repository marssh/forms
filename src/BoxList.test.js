import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BoxList } from './BoxList';

//smoketest

it('testing if page renders', () => {
  render(<BoxList />);
});

it('creating and testing snapshots', () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it('test if inputs change', () => {
  const { getByLabelText } = render(<BoxList />);
  const input = getByLabelText('Background Color');
  
  fireEvent.change(input, { target: { value: 'blue' } });
  expect(input.value).toEqual('blue');
});

it('submit', () => {
  const { getByText, getByLabelText, queryByTestId } = render(<BoxList />)
  const input = getByLabelText('Background Color');
  const height = getByLabelText('Height');
  const width = getByLabelText('Width');
  const button = getByText('Send');

  fireEvent.change(input, { target: { value: 'blue' } });
  fireEvent.change(height, { target: { value: '100' } });
  fireEvent.change(width, { target: { value: '100' } });

  fireEvent.click(button);
  expect(queryByTestId('1')).toBeInTheDocument();
})

it('remove box', () => {
  const { getByText, getByLabelText, queryByTestId } = render(<BoxList />)
  // add
  const input = getByLabelText('Background Color');
  const height = getByLabelText('Height');
  const width = getByLabelText('Width');
  const button = getByText('Send');

  fireEvent.change(input, { target: { value: 'blue' } });
  fireEvent.change(height, { target: { value: '100' } });
  fireEvent.change(width, { target: { value: '100' } });

  fireEvent.click(button);
  //end add

  const deleteButton = getByText('X');
  fireEvent.click(deleteButton);
  expect(queryByTestId('1')).not.toBeInTheDocument();
 
})