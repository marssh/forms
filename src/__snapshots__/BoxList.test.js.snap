// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating and testing snapshots 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        value="0"
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        value="0"
      />
      <button>
        Send
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`submit 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        value="0"
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        value="0"
      />
      <button>
        Send
      </button>
    </form>
    <div
      id="2dca9fc2-ccde-4010-98aa-d7944cd2367a"
    >
      <div
        style="background-color: blue; height: 100px; width: 100px;"
      />
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
